types/bluebird/index.d.ts

```diff
@@ -37,25 +37,32 @@
 
 type CatchFilter<E> = (new (...args: any[]) => E) | ((error: E) => boolean) | (object & E);
 type IterableItem<R> = R extends Iterable<infer U> ? U : never;
-type IterableOrNever<R> = R extends Iterable<any> ? R : never;
+type IterableOrNever<R> = Extract<R, Iterable<any>>;
+type Resolvable<R> = R | PromiseLike<R>;
+type ExtractFunction<T> = Extract<T, (...args: any[]) => any>;
```

Comment:
It only follows fairly the [`common-mistakes`](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes) guidelines. If there is no particular reason, it will be correct to follow the standard.

Reply:
I believe this is something different in this case though.
`Function` is as far as I know _exactly_ the same as writing `(...args: any[]) => any`.
What the "Common Mistakes" guidelines are trying to say instead is that you should try to use specific typings like `(...args: number[]) => string` instead.

`Extract<T, Function>` stems from the TypeScript 2.8 release notes, which is why this should be the preferred solution, since it's shorter and easier to understand.
See the `Extract` and `Exclude` examples [here](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-8.html#example-4).

Reply:
I tried to apply it based on your opinion, but I got a runtime error.
This means that the actual state of `ReturnType<T>` is `type ReturnType <T extends (... args: any []) => any> = T extends (... args: any []) => infer R ? any;` And it seems to have to be `(... args: any []) => any` to retrieve it in `infer`.
Although it is possible to accept with `Function`, it is inevitable that the prospects become worse than do it directly!